kube/registry_credentials.go

```diff
@@ -0,0 +1,38 @@
+package kube
+
+import (
+	"github.com/SUSE/fissile/helm"
+)
+
+// MakeRegistryCredentials generates a template that contains Docker Registry credentials
+func MakeRegistryCredentials(settings *ExportSettings) (helm.Node, error) {
+
+	value := ""
+	if settings.CreateHelmChart {
+		// Registry secrets are in json format:
+		// {
+		//  "docker.io": {
+		//      "username": "foo",
+		//      "password": "bar",
+		//      "auth": "Zm9vOmJhcg=="
+		//   }
+		// }
+		//
+		// where "auth" is a base64 encoded "username:password"
+
+		value = `{{ printf "{%q:{%q:%q,%q:%q,%q:%q}}" ` +
+			`.Values.kube.registry.hostname ` +
+			`"username" (default "" .Values.kube.registry.username) ` +
+			`"password" (default "" .Values.kube.registry.password) ` +
+			`"auth" (printf "%s:%s" .Values.kube.registry.username .Values.kube.registry.password | b64enc) ` +
+			`| b64enc }}`
```

Comment:
Hmm, that is an interesting point. It means we are always assuming "Basic Auth". Maybe we need a mechanism to specify the `auth` credentials in case the registry uses some token auth scheme instead?

As for using `printf` on the literal strings, it just avoids having to escape the doublequotes, I guess. 😄  (They are being printed with `%q` format).

Reply:
FWIW, I could not figure out how to use `imagePullSecrets` with token auth. Well, maybe if the token is not expiring, then you could just login with `docker` and then "steal" the token.

Otherwise you would need to somehow deal with refresh tokens.

All references to `imagePullSecrets` I could find seem to be using basic auth.